closure/private/java_proto_library.bzl

```diff
@@ -0,0 +1,76 @@
+# -*- mode: python; -*-
+#
+# Copyright 2016 The Closure Rules Authors. All rights reserved.
+# Copyright 2014 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _impl(ctx):
+  out = ctx.outputs.srcjar
+  ctx.action(
+      command=' '.join([
+          "JAR='%s'" % ctx.executable._jar.path,
+          "OUTPUT='%s'" % out.path,
+          "PROTO_COMPILER='%s'" % ctx.executable._proto_compiler.path,
+          "SOURCE='%s'" % ctx.file.src.path,
+          ctx.executable._gensrcjar.path,
+      ]),
+      inputs=([ctx.file.src] + ctx.files._gensrcjar + ctx.files._jar +
+              ctx.files._jdk + ctx.files._proto_compiler),
+      outputs=[out],
+      mnemonic="GenProtoSrcJar",
+      use_default_shell_env=True)
+  return struct(runfiles=ctx.runfiles(collect_default=True))
+
+_gensrcjar = rule(
+    implementation=_impl,
+    attrs = {
+        "src": attr.label(
+            allow_files = FileType([".proto"]),
+            single_file = True),
```

Comment:
Not at the moment. This is a private scrap rule mostly copied in from the Bazel codebase. It's good enough for our needs. Provided there aren't any glaring bugs.